main.go

```diff
@@ -355,6 +356,31 @@ func main() {
 		p, err = plural.NewPluralProvider(cfg.PluralCluster, cfg.PluralProvider)
 	case "tencentcloud":
 		p, err = tencentcloud.NewTencentCloudProvider(domainFilter, zoneIDFilter, cfg.TencentCloudConfigFile, cfg.TencentCloudZoneType, cfg.DryRun)
+	case "plugin":
+		startedChan := make(chan struct{})
+		awsProvider, awsErr := aws.NewAWSProvider(aws.AWSConfig{
```

Comment:
should this configuration be moved below (line 379) https://github.com/kubernetes-sigs/external-dns/pull/3063/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R379

Reply:
Yikes, yes. Moving.